service/controller/resource/blobobject/desired.go

```diff
@@ -25,11 +27,78 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	var clusterCerts certs.Cluster
+	var masterCertFiles []certs.File
+	var workerCertFiles []certs.File
 	{
-		clusterCerts, err = r.certsSearcher.SearchCluster(key.ClusterID(&cr))
-		if err != nil {
-			return nil, microerror.Mask(err)
+		g := &errgroup.Group{}
+		m := sync.Mutex{}
+
+		g.Go(func() error {
+			tls, err := r.certsSearcher.SearchTLS(key.ClusterID(&cr), certs.APICert)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			m.Lock()
+			masterCertFiles = append(masterCertFiles, certs.NewFilesAPI(tls)...)
+			m.Unlock()
+
+			return nil
+		})
+
+		g.Go(func() error {
+			tls, err := r.certsSearcher.SearchTLS(key.ClusterID(&cr), certs.CalicoEtcdClientCert)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			m.Lock()
+			masterCertFiles = append(masterCertFiles, certs.NewFilesCalicoEtcdClient(tls)...)
+			workerCertFiles = append(workerCertFiles, certs.NewFilesCalicoEtcdClient(tls)...)
+			m.Unlock()
+
+			return nil
+		})
+
+		g.Go(func() error {
+			tls, err := r.certsSearcher.SearchTLS(key.ClusterID(&cr), certs.EtcdCert)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			m.Lock()
+			masterCertFiles = append(masterCertFiles, certs.NewFilesEtcd(tls)...)
+			m.Unlock()
+
+			return nil
+		})
+
+		g.Go(func() error {
+			tls, err := r.certsSearcher.SearchTLS(key.ClusterID(&cr), certs.ServiceAccountCert)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			m.Lock()
+			masterCertFiles = append(masterCertFiles, certs.NewFilesServiceAccount(tls)...)
+			m.Unlock()
+
+			return nil
+		})
+
+		g.Go(func() error {
+			tls, err := r.certsSearcher.SearchTLS(key.ClusterID(&cr), certs.WorkerCert)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			m.Lock()
+			workerCertFiles = append(workerCertFiles, certs.NewFilesWorker(tls)...)
```

Comment:
Would be nice if someone could double check if the certs are all there.

Reply:
We'll get to it in few days in general. We are at the point of stitching together different components right now in order to jumpstart actual node pool cluster anytime soon.